channeldb/witness_cache_test.go

```diff
@@ -107,7 +107,7 @@ func TestWitnessCacheUnknownWitness(t *testing.T) {
 
 	// We'll attempt to add a new, undefined witness type to the database.
 	// We should get an error.
-	err = wCache.AddWitness(234, key[:])
+	err = wCache.AddWitnesses(234, key[:])
 	if err != ErrUnknownWitnessType {
 		t.Fatalf("expected ErrUnknownWitnessType, got %v", err)
 	}
```

Comment:
Do you want to test adding an actual batch of preimages?

Reply:
indeed! i've also added a test in `link_test.go` to assert the link's behavior is correct